pkg/tfgen/docs.go

```diff
@@ -820,24 +820,89 @@ func (p *tfMarkdownParser) parseSchemaWithNestedSections(subsection []string) {
 }
 
 type markdownLineInfo struct {
-	name, desc          string
-	isFound, isIndented bool
+	name, desc string
+	isFound    bool
+}
+
+type bulletListEntry struct {
+	name  string
+	index int
+}
+
+// trackBulletListIndentation looks at the index of the bullet list marker ( `*`, `-` or `+`) in a docs line and
+// compares it to a collection that tracks the level of list nesting by comparing to the previous list entry's nested
+// level (if any).
+// Note that this function only looks at the placement of the bullet list marker, and assumes same-level list markers
+// to be in the same location in each line. This is not necessarily the case for Markdown, which considers a range of
+// locations within 1-4 whitespace characters, as well as considers the start index of the text following the bullet
+// point. If and when this becomes an issue during docs parsing, we may consider adding some of those rules here.
+// Read more about nested lists in GitHub-flavored Markdown:
+// https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#nested-lists
+//
+//nolint:lll
+func trackBulletListIndentation(line, name string, tracker []bulletListEntry) []bulletListEntry {
+	listMarkerRegex := regexp.MustCompile("[-*+]")
+	listMarkerIndex := listMarkerRegex.FindStringIndex(line)[0]
+
+	// If our tracker is empty, we are at list nested level 0.
+	if len(tracker) == 0 {
+		newEntry := bulletListEntry{
+			name:  name,
+			index: listMarkerIndex,
+		}
+		tracker = append(tracker, newEntry)
+		return tracker
+	}
+	// Always compare to last entry in tracker
+	lastListEntry := tracker[len(tracker)-1]
+
+	// if current line's listMarkerIndex is greater than the tracker's last entry's listMarkerIndex,
+	// make a new tracker entry and push it on there with all the info.
+	if listMarkerIndex > lastListEntry.index {
+		name = lastListEntry.name + "." + name
+		newEntry := bulletListEntry{
+			name:  name,
+			index: listMarkerIndex,
+		}
+		tracker = append(tracker, newEntry)
+		return tracker
+	}
+	// if current line's listMarkerIndex is the same as the last entry's, we're at the same level.
+	if listMarkerIndex == lastListEntry.index {
+		// Replace the last entry in our tracker
+		replaceEntry := bulletListEntry{
+			index: listMarkerIndex,
+		}
+		if len(tracker) == 1 {
+			replaceEntry.name = name
+
+		} else {
+			// use the penultimate entry name to build current name
+			replaceName := tracker[(len(tracker)-2)].name + "." + name
+			replaceEntry.name = replaceName
+		}
+
+		tracker = tracker[:len(tracker)-1]
+		tracker = append(tracker, replaceEntry)
+		return tracker
+	}
+
+	// The current line's listMarkerIndex is smaller that the previous entry's.
+	// Pop off the latest entry, and retry to see if the next previous entry is a match.
+	tracker = tracker[:len(tracker)-1]
+	tracker = trackBulletListIndentation(line, name, tracker)
+	return tracker
```

Comment:
nit: I think this is clearer when it's shorter.
```suggestion
	return trackBulletListIndentation(line, name, tracker[:len(tracker)-1])
```